pkg/controller/direct/gkehub/featuremembership_controller.go

```diff
@@ -153,6 +153,25 @@ func (a *gkeHubAdapter) Find(ctx context.Context) (bool, error) {
 		return false, fmt.Errorf("getting feature %q: %w", a.featureID, err)
 	}
 	a.actual = feature
+
+	// Build the membershipID(in format projects/*/locations/*/memberships/*) of the target membership.
+	// We don't need to resolve the GKEHubMembership resource. Instead, we'll build a membershipID that matches the
+	// membershipIDs in the GKEHubFeature's MembershipSpecs list. This allows us to manage the target membership
+	// using the GKEHubFeatureMembership configuration, without needing to resolve the resource.
+	membershipRef := a.desired.Spec.MembershipRef
+	if membershipRef.External != "" {
+		// use the provided external value
+		a.membershipID = membershipRef.External
+	} else {
+		// build membershipID by its project, location and name
+		if membershipLocation := a.desired.Spec.MembershipLocation; membershipLocation != nil {
+			a.membershipID = fmt.Sprintf("projects/%s/locations/%s/memberships/%s", a.projectID, common.ValueOf(membershipLocation), membershipRef.Name)
```

Comment:
KCC resource name != GCP name, e.g. if spec.resourceID is set.  Does GKEHubFeatureMembership support resourceID?